contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -1328,6 +1341,148 @@ pltsql_post_parse_analyze(ParseState *pstate, Query *query, JumbleState *jstate)
 	}
 }
 
+static bool
+checkForJsonAuto(Query *query)
+{
+	// Detect JSON AUTO
+	List* target = query->targetList;
+	bool isJsonAuto = false;
+	List* rtable;
+	RangeTblEntry* rte;
+	Query* subq;
+	List* subqRtable;
+	RangeTblEntry* subqRte;
+	RangeTblEntry* queryRte;
+	Alias *colnameAlias;
+	int numTables = 0;
+	int currTables = 0;
+	int currMax = 0;
+	forjson_table **tableInfoArr;
+	if(target) {
+		ListCell* lc = list_nth_cell(target, 0);
+		if(lc != NULL && ((Node*) lfirst(lc))->type == T_TargetEntry) {
```

Comment:
Please use nodeTag(node) == T_TargetEntry in case to get the type to make the code more readable